test/integration/disruption/disruption_test.go

```diff
@@ -186,6 +184,8 @@ func TestPDBWithScaleSubresource(t *testing.T) {
 }
 
 func TestEmptySelector(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
this should be by test-case. Moving it to the top doesn't make it clearer, potentially just leads to more noise in logs.